ShopperInsights/src/main/java/com/braintreepayments/api/EligiblePaymentsApi.kt

```diff
@@ -1,28 +1,34 @@
 package com.braintreepayments.api
 
 import com.braintreepayments.api.EligiblePaymentsApiRequest.Companion.toJson
+import org.json.JSONException
 
-internal class EligiblePaymentsApi {
-    fun execute(request: EligiblePaymentsApiRequest): EligiblePaymentsApiResult {
-        request.toJson()
-        // TODO: Network call
-
-        // Hardcoded result
-        return EligiblePaymentsApiResult(
-            eligibleMethods = EligiblePaymentMethods(
-                paypal = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                ),
-                venmo = EligiblePaymentMethodDetails(
-                    canBeVaulted = true,
-                    eligibleInPayPalNetwork = true,
-                    recommended = true,
-                    recommendedPriority = 1
-                )
-            )
+internal class EligiblePaymentsApi(
+    private val braintreeClient: BraintreeClient
+) {
+    fun execute(request: EligiblePaymentsApiRequest, callback: EligiblePaymentsCallback) {
+        val jsonBody = request.toJson()
+        // TODO: Move url to PaypalHttpClient class when it is created
+        val url = "https://api.sandbox.paypal.com/v2/payments/find-eligible-methods"
+        braintreeClient.sendPOST(
+            url,
+            jsonBody,
+            object : HttpResponseCallback {
+                override fun onResult(responseBody: String?, httpError: Exception?) {
```

Comment:
Do we want to do something with the `httpError`? We might want to pipe it through the callback if it's not null.

Reply:
Fixed. bbe66c375bb173fcdf315fdfe5064f93ccba8fd3